### PR TITLE
Ruff changes, expand pre-commit hooks to checkPot and unit tests

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Fix up for #16751 

### Summary of the issue:
- Ruff recently updated, so we might as well use the latest version.
- a warning is omitted for the section we added `logger-objects` to in ruff's config.
- it was unclear in docs how to avoid pre-commit hooks from triggering.
- more pre-commit hooks could be added
- ruff whitespace formating is done through `ruff format` currently we are only linting with `ruff check`. We should also run a `ruff format` on the whole NVDA.
- ruff was missing scons files, which should also be linted in a mass lint.

### Summary of other changes:
- Added pre commit hooks for unit tests and translation comment checks.
- Added generic pre-commit hooks for python static syntax checking, basic whitespace rules, and file write safety checks
- change the output of rununittests.bat to have simpler buffer output, only show failures. Ensure the verbose log is logged in AppVeyor.
- add ruff format to lint checks
- add scons files to lint checks
- update coderabbit prompt for change log entry items

@coderabbitai summary

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
-  rev: v0.4.10
+  rev: v0.5.0
   hooks:
     # Run the linter.
     - id: ruff
       args: [ --fix ]
     # Run the formatter.
     - id: ruff-format
+
+- repo: local
```
Comment (LeonarddeR) [inline .pre-commit-config.yaml:12]: If the intention of this pr is expanding pre-commit hooks, I'd like to suggests additional hooks as used by the nvda dd-on template:

```suggestion
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.6.0
    hooks:
      - id: check-ast
      - id: check-case-conflict
      - id: check-merge-conflict
      - id: end-of-file-fixer
- repo: local
```

Comment (seanbudd) [reply .pre-commit-config.yaml:12]: I think maybe we should also consider `trailing-whitespace`.

Thanks for catching these


```suggestion
- repo: https://github.com/pre-commit/pre-commit-hooks
  rev: v4.6.0
  hooks:
    - id: check-ast
    - id: check-case-conflict
    - id: check-merge-conflict
    - id: end-of-file-fixer
    - id: trim-trailing-whitespace

- repo: local
```

File: pyproject.toml [hunk 2]
```diff
@@ -16,8 +16,6 @@ builtins = [
 	"npgettext",
 ]
 
-logger-objects = ["logHandler.log"]
-
 include = [
 	"*.py",
 	"*.pyw",
```
Comment (LeonarddeR) [inline pyproject.toml:21]: Add the time of flake8, we also linted scons scripts.

```suggestion
	"*.pyw",
	"sconstruct",
	"*sconscript",
```

This needs testing though, probably also a fixup commit to prelint these

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,10 +1,36 @@
...
-  rev: v0.4.10
+  # Matches Ruff version in requirements.
+  rev: v0.5.0
   hooks:
-    # Run the linter.
     - id: ruff
+      name: lint with ruff
       args: [ --fix ]
-    # Run the formatter.
     - id: ruff-format
```
Comment (LeonarddeR) [inline .pre-commit-config.yaml:20]: I think this is too drastic when formatting isn't applied to the code base first. I just tried to commit speech.py but it reformats the whole file.

Comment (seanbudd) [reply .pre-commit-config.yaml:20]: I'll be performing a full `ruff format` in another commit soon

File: rununittests.bat [hunk 4]
```diff
@@ -7,4 +7,4 @@ set unitTestsPath=%here%\tests\unit
 set testOutput=%here%\testOutput\unit
 md %testOutput%
 
-call "%scriptsDir%\venvCmd.bat" py -m xmlrunner discover -v -s "%unitTestsPath%" -t "%here%" --output-file "%testOutput%\unitTests.xml" %*
+call "%scriptsDir%\venvCmd.bat" py -m xmlrunner discover -b -s "%unitTestsPath%" -t "%here%" --output-file "%testOutput%\unitTests.xml" %*
```
Comment (gerald-hartig) [inline rununittests.bat:10]: Why the change from -v to -b?

**Review by gerald-hartig — APPROVED** [review]

File: rununittests.bat (hunk 4 above)
Comment (seanbudd) [reply rununittests.bat:10]: the test success output is incredibly noisy, to the point where it cannot be parsed reasonably. `-b` only shows test failures, and makes it easier to parse for both sighted and screen reader devs. the buffer output for screen reader users is F for each test failure, and without supressing punctuation, a "dot" for each test success. 
we use `-v` for appveyor to retain a full log.

File: pyproject.toml [hunk 5]
```diff
@@ -16,11 +16,11 @@ builtins = [
...
+	"sconstruct",
+	"*sconscript",
```
Comment (LeonarddeR) [inline pyproject.toml:23]: Ugh, I now notice that this gets really noisy. We should either tell Ruff about scons internals like `Return`, `Import`, etc. or revert this, I think.

Comment (seanbudd) [reply pyproject.toml:23]: This seems fixable with 

```
[tool.ruff.lint.per-file-ignores]
# sconscripts contains many inbuilt functions not recognised by the lint,
# so ignore F821.
"sconstruct" = ["F821"]
"*sconscript" = ["F821"]
```
